src/main/java/org/prgrms/kdtspringvoucher/voucher/repository/MemoryVoucherRepository.java

```diff
@@ -0,0 +1,37 @@
+package org.prgrms.kdtspringvoucher.voucher.repository;
+
+import org.prgrms.kdtspringvoucher.voucher.service.Voucher;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Repository;
+
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Repository
+@Profile({"dev"})
+public class MemoryVoucherRepository implements VoucherRepository{
+    private final Map<UUID, Voucher> storage = new ConcurrentHashMap<>();
+
+    @Override
+    public Voucher findById(UUID voucherId) {
+        Optional<Voucher> voucher = Optional.ofNullable(storage.get(voucherId));
+        if (voucher.isEmpty()) {
+            return null;
+        }
+        return voucher.get();
+    }
```

Comment:
이렇게 구현할거면 Optional을 사용할 필요가 없는 것 같아요
어차피 stroage.get(voucherId)가 존재하지 않으면 null이 리턴될거고요
그렇기 때문에 Optional을 사용하면 Optional을 이용하는 방향으로 구현했으면 좋겠어요
그리고 저런 형태는 안티패턴이거든요 Optional을 사용해서 voucher.isEmpty로 찾을거면 vocher == null과 동일하기 때문에요
```suggestion
    public Voucher findById(UUID voucherId) {
        Optional<Voucher> voucher = Optional.ofNullable(storage.get(voucherId));
        return vouhcer.orElse(null)
    }
```
이런표현이 더 좋겠네요